front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -256,9 +302,11 @@ export const usePathfinding = (
               // We update existing pathsteps with coordinates, positionOnPath and kp corresponding to the new pathfinding result
               const updatedPathSteps: (PathStep | null)[] = pathSteps.map((step, i) => {
                 if (!step) return step;
+                const trigramValue = 'trigram' in step ? step.trigram : 'no trigram found';
                 const correspondingOp = suggestedOperationalPoints.find(
                   (suggestedOp) =>
-                    'uic' in step && suggestedOp.uic === step.uic && suggestedOp.ch === step.ch
+                    ('uic' in step && suggestedOp.uic === step.uic && suggestedOp.ch === step.ch) ||
+                    (suggestedOp.trigram === trigramValue && suggestedOp.ch === step.ch)
```

Comment:
Instead of adding more cases in this function, could we use `matchPathStepAndOp()`?

Reply:
Indeed ! Much clearer with the function. Thanks ;-)